src/runtime/server/middleware/session/index.ts

```diff
@@ -129,17 +130,23 @@ const ensureSession = async (event: H3Event) => {
 }
 
 export default eventHandler(async (event: H3Event) => {
+  const sessionOptions = useRuntimeConfig().session.session as SessionOptions
+
   // 1. Ensure that a session is present by either loading or creating one
-  await ensureSession(event)
+  const session = await ensureSession(event)
+  // 2. Save current state of the session
+  const source = { ...session }
 
-  // 2. Setup a hook that saves any changed made to the session by the subsequent endpoints & middlewares
+  // 3. Setup a hook that saves any changed made to the session by the subsequent endpoints & middlewares
   event.res.on('finish', async () => {
     // Session id may not exist if session was deleted
     const session = await getSession(event)
     if (!session) {
       return
     }
 
-    await setStorageSession(session.id, event.context.session)
+    if (sessionOptions.resave || !equal(event.context.session, source)) {
```

Comment:
This conflicts with #36, as the updated `createdAt` would not be saved.
So this should be updated when #36 is merged.
```suggestion
    if (sessionOptions.resave || sessionOptions.rolling || !equal(event.context.session, source)) {
```

Reply:
As this is not straightforward, I would suggest to merge #48 instead of this and #41 individually.

Reply:
Hey @interpretor, here you suggest to merge #48 and #41 (and abandon current one #47), while in https://github.com/sidebase/nuxt-session/pull/48#issuecomment-1332037340 you suggest to merge #36, #41 and #47.

Can you please bring your reasoning on those changes and suggest from which one should I start a review? Perhaps  #36 or #41?

Reply:
Hey @valiafetisov, I think you misunderstood me. Here I suggest to merge #48, as it is based on #36, #41 and #47. #36 is already merged, and I don't recommend to merge #41 or #47 individually. Instead, I recommend to merge #48 as it does handle the options from #41 and #47 better.

So I recommend to start the review from #48.